### PR TITLE
Improve setup_cast

### PR DESCRIPTION
**Pull request by theychx**
I've moved the inactivity check to setup_cast where it belongs, and also made setup_cast less clunky.

### CODE REVIEW

File: catt/controllers.py [hunk 1]
```diff
@@ -211,10 +209,7 @@ def __init__(self, current_state, states, invert=False, fail=False):
         self._state_event = threading.Event()
         self._current_state = current_state
         if self._current_state in self._states_waited_for:
-            if fail:
-                raise ListenerError("Condition is already met (fail is set)")
```
Comment (theychx) [inline catt/controllers.py:215]: (unrelated to the PR)
I've removed this strange bit of code, as it isn't used anywhere. I honestly don't remember why I put it there.

Comment (skorokithakis) [reply catt/controllers.py:215]: I feel you, I hate bits of code whose purpose I don't remember 😛 That's why I usually leave comments anywhere that's not completely obvious.

Comment (theychx) [reply catt/controllers.py:215]: I'm pretty sure it's never been used. I must have convinced myself that it would be useful at some point.

**Comment by theychx** [comment]
So, uhm, what do you thnik of the PR? :smiley:

**Comment by skorokithakis** [comment]
I quite like it! Sorry, I didn't have time to review thoroughly until now, I just got to the PC. Approved!

**Review by skorokithakis — APPROVED** [review]

**Pull request merged**